SPECS/tdnf/virtual-repo-snapshot.patch

```diff
@@ -1133,3 +1133,212 @@ index 0558611..18c84fc 100644
+@@ -92,7 +97,7 @@ _tdnf()
+ {
+     local c=0 cur __opts __cmds
+     COMPREPLY=()
+-    __opts="--assumeno --assumeyes --cacheonly --debugsolver --disableexcludes --disableplugin --disablerepo --downloaddir --downloadonly --enablerepo --enableplugin --snapshottime --exclude --installroot --noautoremove --nogpgcheck --noplugins --quiet --reboot --refresh --releasever --repo --repofrompath --repoid --rpmverbosity --security --sec --setopt --skip --skipconflicts --skipdigest --skipsignature --skipobsoletes --testonly --version --available --duplicates --extras --file --installed --whatdepends --whatrequires --whatenhances --whatobsoletes --whatprovides --whatrecommends --whatrequires --whatsuggests --whatsupplements --depends --enhances --list --obsoletes --provides --recommends --requires --requires --suggests --source --supplements --arch --delete --download --download --gpgcheck --metadata --newest --norepopath --source --urls"
++    __opts="--assumeno --assumeyes --cacheonly --debugsolver --disableexcludes --disableplugin --disablerepo --downloaddir --downloadonly --enablerepo --enableplugin --exclude --excludesnapshot --installroot --noautoremove --nogpgcheck --noplugins --quiet --reboot --refresh --releasever --repo --repofrompath --repoid --rpmverbosity --security --sec --setopt --skip --skipconflicts --skipdigest --skipsignature --skipobsoletes --snapshottime --testonly --version --available --duplicates --extras --file --installed --whatdepends --whatrequires --whatenhances --whatobsoletes --whatprovides --whatrecommends --whatrequires --whatsuggests --whatsupplements --depends --enhances --list --obsoletes --provides --recommends --requires --requires --suggests --source --supplements --arch --delete --download --download --gpgcheck --metadata --newest --norepopath --source --urls"
+     __cmds="autoerase autoremove check check-local check-update clean distro-sync downgrade erase help history info install list makecache mark provides whatprovides reinstall remove repolist repoquery reposync search update update-to updateinfo upgrade upgrade-to"
+     cur="${COMP_WORDS[COMP_CWORD]}"
+     _tdnf__process_if_prev_is_option && return 0
+diff --git a/include/tdnftypes.h b/include/tdnftypes.h
+index f371430..b95c81f 100644
+--- a/include/tdnftypes.h
++++ b/include/tdnftypes.h
+@@ -276,6 +276,7 @@ typedef struct _TDNF_CONF
+ typedef struct _TDNF_REPO_DATA
+ {
+     int nEnabled;
++    int nExcludeSnapshot;
+     int nSkipIfUnavailable;
+     int nGPGCheck;
+     int nHasMetaData;
+diff --git a/tools/cli/lib/help.c b/tools/cli/lib/help.c
+index 86c34e7..4097203 100644
+--- a/tools/cli/lib/help.c
++++ b/tools/cli/lib/help.c
+@@ -23,8 +23,8 @@ static const char *help_msg =
+  "           [--downloadonly]\n"
+  "           [--enablerepo=<repoid>]\n"
+  "           [--enableplugin=<plugin_name>]\n"
+- "           [--snapshottime=<POSIX_time>]\n"
+  "           [--exclude [file1,file2,...]]\n"
++ "           [--excludesnapshot=<repoid>]\n"
```

Comment:
can we rename this to something like --snapshotexclusionrepos so that it sorts with snapshottime.  And helps clarify that the snapshot exclusion applies to a repo.

Also...how big of a stretch from here to take a comma separated list of repos (I know we discussed supporting only 1)

Reply:
Updated to `--snapshotexcluderepos` to make the names come together.

I also added the ability to specify multiple repos in one go as a comma separated list in line.

Reply:
tested by running the following commands to check if the second repo was snapshot when using the updated tdnf:
`tdnf install valgrind --snapshottime=1724119509 --snapshotexcluderepos=azurelinux-official-base`
`tdnf install valgrind --snapshottime=1724119509 --snapshotexcluderepos=azurelinux-official-base,azurelinux-official-base-debug`
![image](https://github.com/user-attachments/assets/7d8fe8bc-2f2e-4e6c-b846-f63fddef847c)